pandas/tests/indexes/timedeltas/test_ops.py

```diff
@@ -1282,3 +1282,23 @@ def test_add_overflow(self):
         result = (to_timedelta([pd.NaT, '5 days', '1 hours']) +
                   to_timedelta(['7 seconds', pd.NaT, '4 hours']))
         tm.assert_index_equal(result, exp)
+
+    def test_timedeltaindex_add_timestamp_nat_masking(self):
+        # GH17991 checking for overflow-masking with NaT
+        tdinat = pd.to_timedelta(['24658 days 11:15:00', 'NaT'])
+
+        tsneg = Timestamp('1950-01-01')
+        ts_neg_variants = [tsneg,
+                           tsneg.to_pydatetime(),
+                           tsneg.to_datetime64().astype('datetime64[ns]'),
+                           tsneg.to_datetime64().astype('datetime64[D]')]
+
+        tspos = Timestamp('1980-01-01')
+        ts_pos_variants = [tspos,
+                           tspos.to_pydatetime(),
+                           tspos.to_datetime64().astype('datetime64[ns]'),
+                           tspos.to_datetime64().astype('datetime64[D]')]
+
+        for variant in ts_neg_variants + ts_pos_variants:
+            res = tdinat + variant
+            assert res[1] is pd.NaT
```

Comment:
check sub as well

Reply:
also would check both add/sub for the reverse, e.g. variant + tdinat (and -)

Reply:
might as well assert fully, e.g.

``tm.assert_index_equal(pd.TimedeltaIndex(['NaT', 'NaT']))``

Reply:
Happy to do the full add/sub/radd/rsub matrix... that was kind of what I started out with.   The question becomes where to put it, since arithmetic tests are scattered about.  My preference would be new test modules `test_arithmetic` in each of `indexes.timedeltas`, `indexes.datetimes`, and `indexes.periods` where I can a) put these new tests and b) collect the arithmetic tests that are currently scattered about.

See discussion in #18026, #18036.

But for now I'll just edit the contents of the tests already introduced.

Reply:
> might as well assert fully, e.g.
>
> tm.assert_index_equal(pd.TimedeltaIndex(['NaT', 'NaT']))
 
The first entry is not `NaT`, will not be constant across all of the variants (though the variants could be split into two groups over which it should be unchanging)

Reply:
test_arithmetic sounds like a good name. key is to share code as much as possible, via fixtures / parametrization / inheritence. We ideally want these objects to act as similar as possible, so keeping special cases to a minimum is important.

note before we do this, I think splitting out the tz-aware tests to its own hierarchy should be done.

see https://github.com/pandas-dev/pandas/pull/17583 and #17694

Reply:
> But for now I'll just edit the contents of the tests already introduced.

Actually as I look at this, I'd much rather close out this bug fix and follow up with the Do It Right approach.

Reply:
certainly. bug fixes are good. separate, self-contained refactorings to make things more readable are better!

Reply:
Great.  After the current deluge of clears up, I'll circle back to this.